lib/Cake/View/Helper/FlashHelper.php

```diff
@@ -82,10 +82,15 @@ public function render($key = 'flash', $options = array()) {
 			));
 		}
 
-		$flash = $options + $flash;
 		CakeSession::delete("Message.$key");
-		$flash['key'] = $key;
 
-		return $this->_View->element($flash['element'], $flash);
+		$out = '';
+		foreach ($flash as $message) {
+			$message['key'] = $key;
+			$message = $options + $message;
+			$out .= $this->_View->element($message['element'], $message);
```

Comment:
Right now the formatting is done in the view element and the HTML is looking quite nice by default imo 
![screenshot 2016-11-01 08 57 40](https://cloud.githubusercontent.com/assets/9390869/19896376/59461a20-a011-11e6-96a5-3da2d73d4461.png)